src/core_crypto/guassian_sampler.rs

```diff
@@ -0,0 +1,105 @@
+use rand_distr::{Distribution, Normal};
+
+/// `TruncatedDiscreteGaussian` represents the parameters of a truncated discrete Gaussian distribution with:
+/// * standard deviation `sigma`
+/// * bounds `[-bound, bound]`, where bound is equal to `(6*sigma).round()`
+/// * mean 0
+#[derive(Clone, Debug)]
+struct TruncatedDiscreteGaussian {
+    sigma: f64,
+    bound: i64,
+}
+
+/// TODO: this struct is likely to contain further fields in the future
+#[derive(Clone, Debug)]
+struct GaussianSampler {
+    params: TruncatedDiscreteGaussian,
+}
+
+impl GaussianSampler {
+    /// `new` returns a new `GaussianSampler` with the given parameters starting from the standard deviation `sigma`.
+    pub fn new(sigma: f64) -> Self {
+        assert!(sigma > 0.0, "sigma must be positive");
+        let bound = (6.0 * sigma).round() as i64;
+        Self {
+            params: TruncatedDiscreteGaussian { sigma, bound },
+        }
+    }
+    /// `norm_f64` returns a normally distributed f64 in
+    /// the range [-f64::MAX, f64::MAX], bounds included,
+    /// with standard normal distribution (mean = 0, std_dev = 1).
+    fn norm_f64(&self) -> f64 {
+        let normal = Normal::new(0.0, 1.0).unwrap();
```

Comment:
Why is 1 supplied as std_dev when instead sigma can supplied directly?

Reply:
I implemented as such to mimic the pattern used by lattigo. But if that's ok for you I can safely remove `norm_f64` and sample directly from the gaussian with standard deviation equal to sigma.

Reply:
Fixed at 16199e8